File: frontend/src/services/conversationService.ts
import request from './api';

export const conversationService = {
  listSessions: (kbId: string, params?: { page?: number; size?: number }) =>
    request.get('/conversations/sessions', {
      params: {
        kbId,
        page: params?.page ?? 0,
        size: params?.size ?? 20,
      },
    }),

  getSessionDetail: (kbId: string, sessionId: string) =>
    request.get(`/conversations/sessions/${sessionId}`, {
      params: { kbId },
    }),

  deleteSession: (kbId: string, sessionId: string) =>
    request.delete(`/conversations/sessions/${sessionId}`, {
      params: { kbId },
    }),
};

File: src/main/java/com/mtmn/smartrag/service/ConversationService.java
package com.mtmn.smartrag.service;

import com.mtmn.smartrag.dto.ConversationSessionDetailResponse;
import com.mtmn.smartrag.dto.ConversationSessionResponse;
import com.mtmn.smartrag.po.Conversation;
import org.springframework.data.domain.Page;

/**
 * 对话历史服务
 */
public interface ConversationService {

    Page<ConversationSessionResponse> listSessions(Long kbId, Long userId, int page, int size);

    ConversationSessionDetailResponse getSessionDetail(Long kbId, Long userId, String sessionId);

    void deleteSession(Long kbId, Long userId, String sessionId);

    Conversation saveConversation(Conversation conversation);

    String buildHistoryText(Long kbId, Long userId, String sessionId, Integer historyWindow);

    String resolveSessionId(String sessionId);
}
